### DISCUSSION TITLE
fx-load with @target

### DISCUSSION THREAD

**Discussion by timathom** [Jinntec/Fore#211] [Q&A] [answered]
I'm having trouble grokking the usage of `@event` and `@target` on `fx-load` in the [load action demo](https://github.com/Jinntec/Fore/blob/0f5692c09eefc9ceed90555b350b5f0f2781857a/demo/load.html#L88). It doesn't seem intuitive to me that `#second` would need to target `#first` (versus having `#first` dispatch an event to `#second`). Or, why are the `@event` and `@target` attributes needed at all, versus having the load actions just fire in order? Can you help me understand how this is working? Thanks!

**Comment by JoernT** [comment, marked as answer]
Actually first dispatching to second should also work if you prefer that. 

```
<fx-load attach-to="#lazyControl" id="first">
    <template>
        <fx-control ref="item">
            <label>an item:</label>
        </fx-control>
        <fx-output ref="item"></fx-output>
    </template>
    <fx-dispatch name="first-loaded" targetid="second" event="action-performed"></fx-dispatch>
</fx-load>
<fx-load id="second"
         event="first-loaded"
         attach-to="#lazyControl2" delay="400">
    <template>
        <fx-control ref="item2">
            <label>another item:</label>
        </fx-control>
        <fx-output ref="item2"></fx-output>
    </template>
</fx-load>
```


However to explain the demo: 
`event` and `target` are generic attributes all actions inherit.`event` being the event to listen for while `target` is the element the event will fire on. With 'target' you can attach the handler on an element other than it's parent (the default).

So the second load listens for the 'action-performed' event of the first which is fired when that action is done. 

It's a matter of taste which approach you prefer. The original demo is more isolating the single actions from each other 
while your suggestion creates a more imperative forward chain.

Both ways deliver similar result.
